File: io/file_io.py
# Python
import os, time

os.makedirs('py_files', exist_ok=True)
start = time.time()

# Write 1,000 files
for i in range(1000):
    with open(f'py_files/file_{i}.txt', 'w') as f:
        f.write('A' * 1048576)  # 1MB

# Read 1,000 files
for i in range(1000):
    with open(f'py_files/file_{i}.txt', 'r') as f:
        f.read()

print(f"Time: {time.time() - start:.3f}s")

File: io/file_io.js
// Node.js
const fs = require('fs').promises;
const path = require('path');
const { performance } = require('perf_hooks');

async function run() {
    await fs.mkdir('node_files', { recursive: true });
    const start = performance.now();

    // Write 1,000 files
    for (let i = 0; i < 1000; i++) {
        await fs.writeFile(path.join('node_files', `file_${i}.txt`), 'A'.repeat(1048576));
    }

    // Read 1,000 files
    for (let i = 0; i < 1000; i++) {
        await fs.readFile(path.join('node_files', `file_${i}.txt`), 'utf8');
    }

    console.log(`Time: ${(performance.now() - start) / 1000}s`);
}

run();